compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -342,7 +342,7 @@ static_assert_size!(ObligationCauseCode<'_>, 32);
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Lift)]
 pub struct MatchExpressionArmCause<'tcx> {
     pub arm_span: Span,
-    pub semi_span: Option<Span>,
+    pub semi_span: Option<(Span, bool)>,
```

Comment:
please document the magic boolean or use an enum

Reply:
Done.